tests/unit/syncModel/diff/workflow.test.ts

```diff
@@ -0,0 +1,160 @@
+import { describe, expect, it } from "vitest";
+
+import { workflowHandler } from "../../../../src/modules/sync/diff/workflow.ts";
+import { WorkflowSyncModel } from "../../../../src/modules/sync/types/syncModel.ts";
+
+describe("workflowHandler", () => {
+  it("creates a replace operation for every changed property", () => {
+    const source: WorkflowSyncModel = {
+      codename: "workflow_1",
+      name: "workflow 1 renamed",
+      scopes: [{
+        content_types: [{
+          codename: "type_1",
+        }, {
+          codename: "type_2",
+        }],
+        collections: [{
+          codename: "collection_1",
+        }],
+      }, {
+        content_types: [{
+          codename: "type_3",
+        }],
+        collections: [{
+          codename: "collection_2",
+        }],
+      }],
+      steps: [{
+        name: "step 1",
+        codename: "step_1",
+        color: "orange",
+        transitions_to: [],
+        role_ids: [],
+      }, {
+        name: "step 2",
+        codename: "step_2",
+        color: "gray",
+        transitions_to: [{ step: { codename: "step_1" } }],
+        role_ids: [],
+      }],
+      published_step: {
+        name: "Published",
+        codename: "Published",
+        unpublish_role_ids: [],
+        create_new_version_role_ids: [],
+      },
+      archived_step: {
+        name: "Archived",
+        codename: "archived",
+        role_ids: [],
+      },
+    };
+
+    const target: WorkflowSyncModel = {
+      codename: "workflow_1",
+      name: "workflow 1",
+      scopes: [{
+        content_types: [{
+          codename: "type_1",
+        }],
+        collections: [{
+          codename: "collection_1",
+        }],
+      }],
+      steps: [{
+        name: "step 1",
+        codename: "step_1",
+        color: "brown",
+        transitions_to: [],
+        role_ids: [],
+      }],
+      published_step: {
+        name: "Published",
+        codename: "Published",
+        unpublish_role_ids: ["roleid"],
+        create_new_version_role_ids: [],
+      },
+      archived_step: {
+        name: "Archived",
+        codename: "archived",
+        role_ids: [],
+      },
+    };
+
+    const result = workflowHandler(source, target);
+
+    expect(result).toStrictEqual(
+      [
+        {
+          oldValue: "workflow 1",
+          op: "replace",
+          path: "/name",
+          value: "workflow 1 renamed",
+        },
+        {
+          oldValue: "brown",
+          op: "replace",
+          path: "/steps/codename:step_1/color",
+          value: "orange",
+        },
```

Comment:
i don't see scopes here even though they are different :/

Reply:
yep, that has to do with the object handler incorrectly used on arrays. fixed